homeassistant/components/tasmota/__init__.py

```diff
@@ -203,22 +221,23 @@ async def async_setup_device(
 @websocket_api.websocket_command(
     {vol.Required("type"): "tasmota/device/remove", vol.Required("device_id"): str}
 )
-@websocket_api.async_response
-async def websocket_remove_device(
+@callback
+def websocket_remove_device(
     hass: HomeAssistant, connection: ActiveConnection, msg: dict
 ) -> None:
     """Delete device."""
     device_id = msg["device_id"]
-    dev_registry = await hass.helpers.device_registry.async_get_registry()
+    dev_registry = dr.async_get(hass)
 
     if not (device := dev_registry.async_get(device_id)):
         connection.send_error(
             msg["id"], websocket_api.const.ERR_NOT_FOUND, "Device not found"
         )
         return
 
-    for config_entry in device.config_entries:
-        config_entry = hass.config_entries.async_get_entry(config_entry)
+    for config_entry_id in device.config_entries:
+        config_entry = hass.config_entries.async_get_entry(config_entry_id)
+        assert config_entry
         # Only delete the device if it belongs to a Tasmota device entry
         if config_entry.domain == DOMAIN:
             dev_registry.async_remove_device(device_id)
```

Comment:
Should we update this too?

Reply:
The WS API can be dropped, done here: https://github.com/home-assistant/core/pull/67382